### PR TITLE
Update public.py

### PR DESCRIPTION
**Pull request by Cfomodz**
adding missing limit_price param (order_type specified LIMIT was acceptable but no mechanism was in place to attach the desired limit price leading to preflight failure of order)

+ Added the .text of the response if status code is not 200 for debugging/error handling

### CODE REVIEW

**Comment by Cfomodz** [comment]
I don't _mind_ restructuring the function to make DeepSource happy.. but I was trying to make as few changes as possible to the flow considering it's my first contribution to the project 😅

File: README.md [hunk 1]
```diff
@@ -50,6 +50,7 @@ order = public.place_order(
     quantity=1,
     side='BUY', # or 'SELL'
     order_type='MARKET', # or 'LIMIT' or 'STOP'
+    limit_price=None # pass if using 'LIMIT' order_type
```
Comment (NelsonDane) [inline README.md]: I'd make a note that this can be a float/decimal value. Like "pass float if using..."

File: public_invest_api/public.py [hunk 2]
```diff
@@ -243,7 +248,7 @@ def place_order(
         )
         if build_response.status_code != 200:
             print(build_response.text)
-            raise Exception("Build order failed")
+            raise Exception(f"Build order failed - {build_response.text}")
```
Comment (NelsonDane) [inline public_invest_api/public.py]: Use a `:` instead of the dash here to match some of the other Exceptions

**Comment by NelsonDane** [comment]
> I don't _mind_ restructuring the function to make DeepSource happy.. but I was trying to make as few changes as possible to the flow considering it's my first contribution to the project 😅

No worries, the specific issue it was fussing about I usually disabled in other repos too, so it's all good now :)

**Review by NelsonDane — CHANGES_REQUESTED** [review]
Thanks so much for the contribution! Functionality looks fine, just some tiny style things.

**Comment by Cfomodz** [comment]
o7

**Review by NelsonDane — APPROVED** [review]
LGTM

**Pull request merged**